src/transformers/trainer.py

```diff
@@ -3831,6 +3831,16 @@ def evaluation_loop(
             if is_torch_xla_available():
                 xm.mark_step()
 
+            if self.args.eval_use_gather_object and not is_accelerate_available("0.30.0"):
+                logger.warning(
+                    "You are using eval_use_gather_object = True with a version of `accelerate` < 0.30.0. This is not supported"
+                    " and we recommend you to update your version."
+                )
+            if "use_gather_object" in inspect.signature(self.gather_function).parameters.keys():
```

Comment:
This is quite an expensive check to add, especially as it occurs for every step from the dataloader. 

Do we expect people to ever modify `gather_object` after they've created `trainer`? If not, then this should be done once in the initial trainer set-up; otherwise at the start of the evaluation loop

Reply:
@amyeroberts I also think people should not or may not ever modify gather_object after trainer created. I will move this function to the initial part!